### ISSUE TITLE
Allow for callback-less Promise-returning invocation

### ISSUE THREAD

**Issue by deanrad** [meteor/validated-method#9]
Interested in a PR for this ?

**Comment by stubailo** [comment]
Yeah sure - actually I would probably be even more interested in a feature to add mixins to methods.

Could you show me what you envision it looking like to call a method with a returned promise?

**Comment by deanrad** [comment]
``` js
Lists.methods.makePrivate.call({listId: list._id}).then(doSomethingWithResult, handleError);
or
Lists.methods.makePrivate.call({listId: list._id})
  .then(doSomethingWithResult)
  .catch(handleError);
```

^ Promises FTW. I got no opinion or ideas regarding mixins.

**Comment by stubailo** [comment]
So is the idea that you always get a promise if you don't pass a callback? What about when you call the method from the server, where the return value is a real object?

**Comment by deanrad** [comment]
Is the challenge that a method currently has a synchronous value (from the stub), and an async value (handled now via callaback) ? 

So this is a little cumbersome, but does something like this capture all the possibilities?

```
//on client
let {clientValue, promisedServerValue} = Lists.methods.makePrivate.call({listId: list._id});

//on server
let actualServerValue} = Lists.methods.makePrivate.call({listId: list._id});
```

**Comment by stubailo** [comment]
Hmm... so is this better than having a method `callPromise` or `callAsync`?

**Comment by deanrad** [comment]
If those names are conventional across other parts of the platform, I'd say those would be great. 

But am I right about needing to grab client value and server value on the client being part of the issue ?

**Comment by stubailo** [comment]
Yeah the issue is that the return values on the client and server already mean something. On the client, the return value is the stub return value; on the server, it's the actual return value of the method. In both cases, returning a promise instead would significantly change the way the method would need to be called, so it makes sense to me that one would opt-in to the promise approach.

Another question is - how do you get the stub return value if you return a promise? I guess you could do the thing you suggested above with multiple return values, but that doesn't really seem that nice and wouldn't work with `await` syntax.

**Comment by stubailo** [comment]
This could be accomplished via mixin:

https://github.com/meteor/validated-method#mixins
